genome.spec

```diff
@@ -75,7 +94,59 @@ typedef structure {
+  location: list of (string, int, string, int) - list of locations from where
+      this sequence originates in the original assembly. Each sub-sequence in the
+      list constitutes a section of the resulting CDS. The first element in
+      the tuple corresponds to the "contig_id", such as "NC_000913.3". The second
+      element in the tuple is an index in the contig of where the sequence starts.
+      The third element is either a plus or minus sign indicating whether it is on
+      the 5' to 3' leading strand ("+") or on the 3' to 5' lagging strand ("-"). The last element
+      is the length of the sub-sequence.
+      For a location on the leading strand (denoted by "+"), the index is of
+      the leftmost base, and the sequence extends to the right.
+      For a location on the lagging strand (denoted by "-"), the index is of
+      the rightmost base, and the sequence extends to the left.
+      NOTE: the third element in each tuple is the *length* of each
+      sub-sequence. If you have a location such as ("xyz", 100, "+", 50), then
+      your sequence will go from index 100 to index 149 (this has a length of
+      50). It *does not* go from index 100 to index 150, as that would have a
+      length of 51.
+      Likewise, if you have the location ("xyz", 100, "-", 50), then the
+      sequence extends from 100 down to 51, which has a length of 50 bases. It
+      does not go from index 100 to 50, as that would have a length of 51.
+  md5: string - TODO
+  protein_md5: string - hash of the protein sequence that this CDS encodes
+  parent_gene: string - gene (feature) from which this CDS comes from,
+      including introns and UTRs that have been removed to create this CDS.
+  parent_mrna: string - mRNA sequence from which this sequence is derived,
+      including UTRs but not introns.
+  note: string - TODO
+  functions: list of string - TODO list of protein products or chemical
+      processes that sequence creates, facilitates, or influences.
+  functional_descriptions: list of string - TODO list of protein products or chemical
+      processes that sequence creates, facilitates, or influences.
+  ontology_terms: map of string to (map of string to (list of int)) - a mapping
+      of ontology source id (eg. "GO") to a mapping of term IDs (eg "GO:16209") to
+      a list of indexes into the ontology events. The index into an ontology
+      event indicates what service and method created this term assignment.
+  flags: list of string - controlled vocab - fields from the genbank source. A
+      common example is "pseudo" for pseudo-genes that do not encode proteins,
+      which shows up as "/pseudo" in the genbank.
+  warnings: list of string - TODO
+  inference_data: list of InferenceInfo -
+  protein_translation: string - amino acid sequence that this CDS gets translated into.
+  protein_translation_length: int - length of the above
+  aliases: list of (string, string) - alternative list of names or identifiers
```

Comment:
What's the 1st and 2nd element of each tuple? Examples would be good

Reply:
Added clarification + examples

Reply:
Added an example, but I'm not really sure what the keys here are exactly. Another Jason question.